File: data/code_pairs/cpp/1874.cpp
// ELHEX27_TESTE.hpp
#ifndef ELHEX27_TESTE_HPP
#define ELHEX27_TESTE_HPP

#include <vector>

const int P = 2, Q = 2, W = 2, NDS = 3;

class ELHEX27_TESTE {
public:
    static void ELHEX27_TESTE_Function(int NEL, std::vector<std::vector<double>>& KE, 
                                       const std::vector<std::vector<double>>& EL_DDISP, 
                                       std::vector<std::vector<double>>& FINTE, int INC);
};

#endif // ELHEX27_TESTE_HPP

// ELHEX27_TESTE.cpp
#include "ELHEX27_TESTE.hpp"

void ELHEX27_TESTE::ELHEX27_TESTE_Function(int NEL, std::vector<std::vector<double>>& KE, 
                                           const std::vector<std::vector<double>>& EL_DDISP, 
                                           std::vector<std::vector<double>>& FINTE, int INC) {
    // Function logic here
    // For demonstration, this function will not perform any calculations.
    std::fill(FINTE.begin(), FINTE.end(), std::vector<double>(1, 0.0));
}

// main.cpp
#include "ELHEX27_TESTE.hpp"
#include <iostream>

int main() {
    int NEL = 1;
    int INC = 1;
    int dim = (P + 1) * (Q + 1) * (W + 1) * NDS;
    std::vector<std::vector<double>> KE(dim, std::vector<double>(dim, 0.0));
    std::vector<std::vector<double>> EL_DDISP(dim, std::vector<double>(1, 0.0));
    std::vector<std::vector<double>> FINTE(dim, std::vector<double>(1, 0.0));

    ELHEX27_TESTE::ELHEX27_TESTE_Function(NEL, KE, EL_DDISP, FINTE, INC);

    std::cout << "FINTE first element: " << FINTE[0][0] << std::endl;

    return 0;
}

File: data/code_pairs/fortran/1874.f90
! ELHEX27_TESTE__genmod.f90
MODULE ELHEX27_TESTE__genmod
  INTERFACE 
    SUBROUTINE ELHEX27_TESTE(NEL,KE,EL_DDISP,FINTE,INC)
      USE MOD_VARIABLES
      INTEGER(KIND=4) :: NEL
      REAL(KIND=8), INTENT(OUT) :: KE((P+1)*(Q+1)*(W+1)*NDS,(P+1)*(Q+1)*(W+1)*NDS)
      REAL(KIND=8), INTENT(IN) :: EL_DDISP((P+1)*(Q+1)*(W+1)*NDS,1)
      REAL(KIND=8), INTENT(OUT) :: FINTE((P+1)*(Q+1)*(W+1)*NDS,1)
      INTEGER(KIND=4) :: INC
    END SUBROUTINE ELHEX27_TESTE
  END INTERFACE 
END MODULE ELHEX27_TESTE__genmod

! MOD_VARIABLES.f90
MODULE MOD_VARIABLES
  INTEGER, PARAMETER :: P=2, Q=2, W=2, NDS=3
END MODULE

! test_ELHEX27_TESTE.f90
PROGRAM test_ELHEX27_TESTE
  USE ELHEX27_TESTE__genmod
  USE MOD_VARIABLES
  IMPLICIT NONE
  INTEGER :: NEL, INC
  INTEGER, PARAMETER :: dim = (P+1)*(Q+1)*(W+1)*NDS
  REAL(KIND=8), ALLOCATABLE :: KE(:,:), EL_DDISP(:,:), FINTE(:,:)

  NEL = 1
  INC = 1
  ALLOCATE(KE(dim,dim), EL_DDISP(dim,1), FINTE(dim,1))
  
  CALL ELHEX27_TESTE(NEL, KE, EL_DDISP, FINTE, INC)
  
  PRINT *, "FINTE:"
  PRINT *, FINTE

  DEALLOCATE(KE, EL_DDISP, FINTE)
END PROGRAM test_ELHEX27_TESTE